src/EasyCaching.HybridCache/HybridCachingProvider.cs

```diff
@@ -159,6 +159,16 @@ private void OnMessage(EasyCachingMessage message)
             }
         }
 
+        /// <summary>
+        /// On reconnect (flushes local memory as it could be stale.
+        /// </summary>
+
+        private void OnReconnect()
+        {
+            LogMessage("Bus Reconnected, flushing in-memory keys");
+            _localCache.Flush();
```

Comment:
Why flushing in-memory keys here?

Reply:
If a disconnection happens (ie the load balancer between app and rabbit fails) then the queue will be deleted and any invalidations will be missed. I believe the correct approach is to flush the in-memory buffer in order to avoid having stale data.

Reply:
It's not a good idea to flush the in memory caching here!

Reply:
I am not sure that understand. Do you suggest to keep the (possibly stale) in-memory cache as is after a disconnection or do you have in mind a better way to refresh the in-memory cache?

Reply:
>  Do you suggest to keep the (possibly stale) in-memory cache as is after a disconnection

Yes.

When a reconnection occurs, not all memory caches are expired, and if `flush` is executed at this time, it may cause application jitter. 

In addition, all in-memory caches have an expiration time, and they are automatically removed after they expire.

Reply:
Ok, that's an understandable argument. Our use case requires however to be sure that the case is not stale. Would it be ok if we do either of these things:

a) Add an option to FlushOnReconnect (default: false)
b) Add a callback in order for the application to decide how to handle a reconnection 

I would prefer (a) but I can do (b)

Reply:
(a) is a good idea.
(b) is ok as well.

😄